### PR TITLE
Panic on wildcard in Cors builder's allowed_origin()

### PR DESCRIPTION
**Pull request by eupn**
## PR Type

Feature

## PR Checklist
Check your PR fulfills the following:

- [x] Tests for the changes have been added / updated.
- [x] Documentation comments have been added / updated.
- [x] A changelog entry has been made for the appropriate packages.
- [x] Format code with the latest stable rustfmt

## Overview

Implements #112.

### CODE REVIEW

File: actix-cors/CHANGES.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 ## Unreleased - 2020-xx-xx
 
+* Disallow `*` in `Cors::allowed_origin` by panicking. [#112].
```
Comment (robjtede) [inline actix-cors/CHANGES.md:5]:
```suggestion
* Disallow `*` in `Cors::allowed_origin` by panicking. [#114].

[#114]: https://github.com/actix/actix-extras/pull/114

```

File: actix-cors/src/builder.rs [hunk 2]
```diff
@@ -115,10 +115,19 @@ impl Cors {
     /// `allowed_origin_fn` function is set, these functions will be used to determinate
     /// allowed origins.
     ///
-    /// Builder panics if supplied origin is not valid uri.
+    /// # Panics
+    ///
+    /// * If supplied origin is not valid uri, or
```
Comment (robjtede) [inline actix-cors/src/builder.rs:120]: Does it? Line 114 seems to suggest otherwise.

File: actix-cors/src/builder.rs [hunk 3]
```diff
@@ -115,10 +115,19 @@ impl Cors {
...
+    /// * If supplied origin is a wildcard (`*`). [`Cors::send_wildcard`] should be used instead.
     ///
     /// [Fetch Standard]: https://fetch.spec.whatwg.org/#origin-header
     pub fn allowed_origin(mut self, origin: &str) -> Cors {
+        assert!(
+            origin != "*",
+            "Wildcard in allowed_origin is not allowed. \
+        Please use send_wildcard() instead."
```
Comment (robjtede) [inline actix-cors/src/builder.rs:128]:
```suggestion
            "Wildcard in `allowed_origin` is not allowed. Use `send_wildcard`."
```

**Review by robjtede — CHANGES_REQUESTED** [review]
thanks for picking this up :)

just some initial thoughts, will sanity check running it later

File: actix-cors/src/builder.rs (hunk 2 above)
Comment (eupn) [reply actix-cors/src/builder.rs:120]: Sorry I’m not following. How does it?

Comment (robjtede) [reply actix-cors/src/builder.rs:120]: Ahh, I'd forgotten how this works. So it assigns an error to the error field and then `finish` panics if there is an error from anywhere in the process. So in this way it's not *this* method that panics for invalid URLs. Therefore, it should not be documented here (in the same section as panics generated by this method, at least).

Thoughts on that?

Comment (eupn) [reply actix-cors/src/builder.rs:120]: It was there before this PR. Should I remove it?

Comment (robjtede) [reply actix-cors/src/builder.rs:120]: Oh, sorry. Somehow didn't notice that. We can leave it in this PR but I'll likely move it later (probably with some other stuff).

**Review by robjtede — APPROVED** [review]

**Pull request merged**